src/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs

```diff
@@ -145,5 +145,10 @@ private string GetPropertyName(in ReadStackFrame frame)
 
             return propertyName;
         }
+
+        /// <summary>
+        /// Bytes consumed in the current loop
+        /// </summary>
+        public long BytesConsumed;
```

Comment:
Something more descriptive since it is only for the read-ahead. Like `ReadAheadBytesConsumed`

Reply:
We always use it, even when we aren't reading ahead. We had to track it separately because of the Skip logic. I've started a discussion with @ahsonkhan about whether or not we can move this back to the reader as there is currently no way to peek ahead without separately tracking consumed.